spring-xd-dirt/src/main/java/org/springframework/xd/dirt/modules/metadata/JmsSourceModuleOptionsMetadata.java

```diff
@@ -42,6 +44,8 @@
 
 	private String clientId = null;
 
+	private String acknowledge = "auto";
```

Comment:
Yes, the DMLC is only used for `transacted`. The basic difference is that with the SMLC, the message listener is a true JMS `MessageListener` and the ack (auto or dups-ok) is sent when the listener exits.

With the DMLC, `consumer.receive()` is called (with the ack emitted when receive exits) and __then__ the listener is invoked.

If you run the test stream without this commit, you will see that the failed message is never redelivered because it was already ack'd.